isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/spring.adoc

```diff
@@ -0,0 +1,290 @@
+Generell gilt, dass jede zentrale und wichtige Klasse aber als Spring-Bean konfiguriert werden sollte.
+
+Für die Modellierung und Konfiguration der Spring-Beans werden im folgenden Vorgaben aufgestellt.
+
+[[querschnitt-als-beans-konfigurieren]]
+==== Querschnittliche Funktionalität als Spring-Beans konfigurieren
+
+Querschnittliche Funktionalität (etwa für das Monitoring, für Fehler- und Nachrichtentexte oder für die Versendung von Mails) sind als Spring-Beans zu konfigurieren.
+Ebenfalls über Spring-Beans durchzuführen ist die Konfiguration diverser Frameworks, z.B. Hibernate.
+Die Konfiguration dieser Frameworks wird in ihren Nutzungsvorgaben beschrieben.
+
+[[beans-standardmaessig-als-singletons-definieren]]
+==== Beans standardmäßig als Singletons definieren
+
+Spring-Beans können entweder als Singletons mit nur einer Instanz, mit einer Instanz pro Aufruf oder mit einer Instanz pro Abhängigkeit (Prototype) erzeugt werden.
+Die Spring-Beans einer Anwendung sollen zustandslos sein und werden als Singleton-Beans erzeugt.
+Wo technisch erforderlich können auch andere Scopes verwendet werden.
+
+[[vorgaben-zur-spring-konfiguration]]
+==== Vorgaben zur Spring-Konfiguration
+
+Die Spring-Konfiguration der Anwendung ist nach folgenden Vorgaben zu erstellen.
+
+Der zentrale Ausgangspunkt für die Spring-Konfiguration ist die <<listing-applikationsklasse>>.
+Diese wird im Package der Anwendung (`<org>.<domäne>.<anwendung>`) erstellt.
+Sie ist mit `@Configuration` und `@EnableAutoConfiguration` annotiert.
+Damit wird die Autokonfiguration von Spring Boot eingeschaltet.
+Zusätzlich erbt die Klasse von `SpringBootServletInitializer`, damit die Anwendung im Tomcat deployt werden kann.
+Die Konfigurationsklassen der Schichten werden per `@Import`-Annotation eingebunden.
+
+[[listing-applikationsklasse]]
+.Applikationsklasse der Anwendung
+[source,java]
+----
+package de.beispiel.if2anwendung;
+
+@Configuration
+@EnableAutoConfiguration
+@Import({ CoreConfig.class, PersistenceConfig.class, ServiceConfig.class })
+public class IsyFactApplication extends SpringBootServletInitializer {
+
+    @Override
+    protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
+        return application.sources(IsyFactApplication.class);
+    }
+}
+----
+
+Für jede Schicht wird im Package der Schicht (`<org>.<domäne>.<anwendung>.<schicht>`) parallel zu den Interfaces der Komponentenschnittstellen eine mit `@Configuration` annotierte <<listing-konfigurationsklasse,Konfigurationsklasse>> erstellt.
+Diese wird zusätzlich mit `@ComponentScan` annotiert.
+Annotationen, die querschnittliche Aspekte konfigurieren (z.B. `@EnableTransactionManagement`), werden an die Konfigurationsklasse der inhaltlich passenden Schicht geschrieben.
+
+[[listing-konfigurationsklasse]]
+.Konfigurationsklasse des Anwendungskerns
+[source,java]
```

Comment:
Ich hätte in einem eigenen Kapitel mit dem Titel "Konfiguration von Spring" (der Vollständigkeit halber) vier Unterkapitel erwartet
1. Vorgaben zur Konfiguration der Anwendung"
2. Vorgaben zur Konfiguration der Service-Schicht
3. Vorgaben zur Konfiguration des Anwendungskerns
4. Vorgaben zur Konfiguration der Persistenzschicht
5. 
Im Text findet man nur ein paar Details zu 1. und 3.

Reply:
Ist out of scope für dieses Ticket. Ich nehme das Thema mit.